server/server.js

```diff
@@ -35,6 +36,10 @@ app.get('/trees', function (req, res) {
     join = ", certificates, donors, organizations";
     joinCriteria = "AND trees.certificate_id = certificates.id AND certificates.donor_id = donors.id AND donors.organization_id = organizations.id AND organizations.id = " + organization;
     subset = true;
+  } else if (flavor) {
+    join = "INNER JOIN tree_attributes ON tree_attributes.tree_id = trees.id";
+    joinCriteria = "AND tree_attributes.key = 'app_flavor' AND tree_attributes.value = '" + flavor + "'";
+    subset = true;
```

Comment:
I wonder if subset should just be initialized as `token || organization || flavor || treeid` to reduce the copy paste a bit

Reply:
yeah, i was thinking about that, too, how to clean it up. i think the copy-paste makes it easily understandable, but more verbose. could maybe split it out into a few functions, will come up with a proposal.